discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -128,23 +128,23 @@ def set_cluster_name(self, cluster_name):
     def select_installation_disk(self, hosts_with_disk_paths):
         self.api_client.select_installation_disk(self.id, hosts_with_disk_paths)
 
-    def set_ocs(self, ocs_enabled):
-        logging.info(f'Enabling Ocs to:{ocs_enabled} for cluster: {self.id}')
-        self.set_operator('ocs', ocs_enabled)
+    def set_ocs(self, ocs_enabled, properties=None):
+        self.set_operator('ocs', ocs_enabled, properties=properties)
 
-    def set_cnv(self, cnv_enabled):
-        logging.info(f'Enabling CNV to:{cnv_enabled} for cluster: {self.id}')
-        self.set_operator('cnv', cnv_enabled)
-
-    def set_operator(self, operator, enabled):
+    def set_operator(self, operator_name, enabled, properties=None):
+        logging.info(f'Enabling {operator_name} to:{enabled} for cluster: {self.id}')
         cluster = self.api_client.cluster_get(self.id)
-        olm_operators = cluster.olm_operators
-        olm_names = [olm_operator['name'] for olm_operator in olm_operators]
-        if operator not in olm_names and enabled:
-            olm_operators.append({'name': operator})
-        if operator in olm_names and not enabled:
-            olm_operators.remove({'name': operator})
-        self.api_client.update_cluster(self.id, olm_operators)
+
+        olm_operators = []
+        for operator in cluster.monitored_operators:
+            if operator.name == operator_name and not enabled:
+                continue
+            olm_operators.append({'name': operator.name, 'properties': operator.properties})
```

Comment:
Monitored operators include all the operators, even builtins. This should check whether the operator type is OLM